### ISSUE TITLE
Issue building using clang-3.5

### ISSUE THREAD

**Issue by ctcyang** [agency-library/agency#370]
agency branch: master 
version.cpp: Agency v0.2
Ubuntu 14.04.5 LTS (GNU/Linux 3.13.0-106-generic x86_64)
Ubuntu clang version 3.5.0-4ubuntu2~trusty2 (tags/RELEASE_350/final) (based on LLVM 3.5.0)
Cuda compilation tools, release 8.0, V8.0.44

---

I can build the  .cu files in the `agency/example/cuda` and `agency/testing/cuda` folders, but I am running into trouble building some of the .cpps under `agency/example` and `agency/testing`.  When I use command `scons -j16 arch=sm_35` in the example folder, I get

```
scons: Reading SConscript files ...
scons: done reading SConscript files.
scons: Building targets ...
clang-3.5 -o fork_executor.o -c -std=c++11 -O3 -Wall -Wno-unused-private-field -Wextra -Wno-mismatched-tags -Wno-unused-parameter -Werror -I/home/ctcyang/agency fork_executor.cpp
clang-3.5 -o hello_async hello_async.o -lstdc++ -lpthread
clang-3.5 -o hello_then hello_then.o -lstdc++ -lpthread
clang-3.5 -o ping_pong_tournament ping_pong_tournament.o -lstdc++ -lpthread
clang-3.5 -o saxpy saxpy.o -lstdc++ -lpthread
fork_executor.cpp:124:17: error: no member named 'accumulate' in namespace 'std'
    return std::accumulate(this_tile.begin(), this_tile.end(), 0);
           ~~~~~^
fork_executor.cpp:128:15: error: no member named 'accumulate' in namespace 'std'
  return std::accumulate(partial_sums.begin(), partial_sums.end(), 0);
         ~~~~~^
fork_executor.cpp:118:8: error: variable has incomplete type 'void'
  auto partial_sums = agency::bulk_invoke(policy(num_agents), [=](agency::parall...
       ^
fork_executor.cpp:137:20: note: in instantiation of function template specialization
      'parallel_sum<const agency::parallel_execution_policy &>' requested here
  int reference  = parallel_sum(agency::par, vec.data(), vec.size());
                   ^
fork_executor.cpp:118:8: error: variable has incomplete type 'void'
  auto partial_sums = agency::bulk_invoke(policy(num_agents), [=](agency::parall...
       ^
fork_executor.cpp:142:18: note: in instantiation of function template specialization
      'parallel_sum<agency::basic_execution_policy<agency::detail::basic_execution_agent<agency::parallel_execution_tag,
      unsigned long>, fork_executor, void> >' requested here
  int fork_sum = parallel_sum(agency::par.on(fork_exec), vec.data(), vec.size());
                 ^
4 errors generated.
scons: *** [fork_executor.o] Error 1
scons: building terminated because of errors.
```

For the testing folder .cpp files, I get a series of errors that look like:

```
experimental/ranges/zip.cpp:31:22: error: invalid operands to binary expression
      ('iterator' (aka
      'zip_with_iterator<agency::experimental::detail::forward_as_tuple_functor,
      range_iterator_t<std::vector<int, std::allocator<int> > &>,
      range_iterator_t<std::vector<int, std::allocator<int> > &>,
      range_iterator_t<std::vector<int, std::allocator<int> > &> >') and 'iterator')
    assert(z.begin() == z.begin());
           ~~~~~~~~~ ^  ~~~~~~~~~
/usr/include/assert.h:89:5: note: expanded from macro 'assert'
  ((expr)                                                               \
    ^
experimental/ranges/zip.cpp:39:23: error: invalid operands to binary expression
      ('iterator' (aka
      'zip_with_iterator<agency::experimental::detail::forward_as_tuple_functor,
      range_iterator_t<std::vector<int, std::allocator<int> > &>,
      range_iterator_t<std::vector<int, std::allocator<int> > &> >') and 'iterator')
    assert(z1.begin() != z2.begin());
           ~~~~~~~~~~ ^  ~~~~~~~~~~
/usr/include/assert.h:89:5: note: expanded from macro 'assert'
  ((expr)                                                               \
    ^
```

**Comment by jaredhoberock** [comment]
Thanks for the report. I think the first issue with `fork_executor.cpp` may be corrected now. Let me know if it works for you.

I'm not sure what to make of the second issue. If it helps, the version of clang I'm using on my workstation is 3.8:

```
$ clang --version
clang version 3.8.0-2ubuntu4 (tags/RELEASE_380/final)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
```

**Comment by jaredhoberock** [comment]
I don't see any errors on my end when I compile `zip.cpp` with `clang-3.5`:

```
$ clang-3.5 -std=c++11 -I../../.. -lstdc++ zip.cpp -o zip
$ ./zip
OK
```

Could you post the command line you're using to compile `zip.cpp`?

**Comment by ctcyang** [comment]
Thanks, that fixes my compilation problem with `agency/examples`. Under `agency/testing`, the command I'm using to compile `zip.cpp` is:

```
clang-3.5 -o experimental/ranges/zip.o -c -std=c++11 -O3 -Wall -Wno-unused-private-field -Wextra -Wno-mismatched-tags -Wno-unused-parameter -Werror -I/home/ctcyang/agency experimental/ranges/zip.cpp
```

I tried

```
clang-3.5 -std=c++11 -I../../.. -lstdc++ zip.cpp -o zip
```

as well just now and it gives the same error:

```
zip.cpp:31:22: error: invalid operands to binary expression ('iterator' (aka
      'zip_with_iterator<agency::experimental::detail::forward_as_tuple_functor,
      range_iterator_t<std::vector<int, std::allocator<int> > &>,
      range_iterator_t<std::vector<int, std::allocator<int> > &>,
      range_iterator_t<std::vector<int, std::allocator<int> > &> >') and 'iterator')
    assert(z.begin() == z.begin());
           ~~~~~~~~~ ^  ~~~~~~~~~
/usr/include/assert.h:89:5: note: expanded from macro 'assert'
  ((expr)                                                               \
    ^
zip.cpp:39:23: error: invalid operands to binary expression ('iterator' (aka
      'zip_with_iterator<agency::experimental::detail::forward_as_tuple_functor,
      range_iterator_t<std::vector<int, std::allocator<int> > &>,
      range_iterator_t<std::vector<int, std::allocator<int> > &> >') and 'iterator')
    assert(z1.begin() != z2.begin());
           ~~~~~~~~~~ ^  ~~~~~~~~~~
/usr/include/assert.h:89:5: note: expanded from macro 'assert'
  ((expr)                                                               \
    ^
zip.cpp:44:20: error: invalid operands to binary expression ('sentinel' (aka
      'zip_with_sentinel<agency::experimental::detail::forward_as_tuple_functor,
      range_sentinel_t<std::vector<int, std::allocator<int> > &>,
      range_sentinel_t<std::vector<int, std::allocator<int> > &>,
      range_sentinel_t<std::vector<int, std::allocator<int> > &> >') and 'sentinel')
    assert(z.end() == z.end());
           ~~~~~~~ ^  ~~~~~~~
/usr/include/assert.h:89:5: note: expanded from macro 'assert'
  ((expr)                                                               \
    ^
zip.cpp:49:22: error: invalid operands to binary expression ('iterator' (aka
      'zip_with_iterator<agency::experimental::detail::forward_as_tuple_functor,
      range_iterator_t<std::vector<int, std::allocator<int> > &>,
      range_iterator_t<std::vector<int, std::allocator<int> > &>,
      range_iterator_t<std::vector<int, std::allocator<int> > &> >') and 'sentinel' (aka
      'zip_with_sentinel<agency::experimental::detail::forward_as_tuple_functor,
      range_sentinel_t<std::vector<int, std::allocator<int> > &>,
      range_sentinel_t<std::vector<int, std::allocator<int> > &>,
      range_sentinel_t<std::vector<int, std::allocator<int> > &> >'))
    assert(z.begin() != z.end());
           ~~~~~~~~~ ^  ~~~~~~~
/usr/include/assert.h:89:5: note: expanded from macro 'assert'
  ((expr)                                                               \
    ^
zip.cpp:54:20: error: invalid operands to binary expression ('sentinel' (aka
      'zip_with_sentinel<agency::experimental::detail::forward_as_tuple_functor,
      range_sentinel_t<std::vector<int, std::allocator<int> > &>,
      range_sentinel_t<std::vector<int, std::allocator<int> > &>,
      range_sentinel_t<std::vector<int, std::allocator<int> > &> >') and 'iterator' (aka
      'zip_with_iterator<agency::experimental::detail::forward_as_tuple_functor,
      range_iterator_t<std::vector<int, std::allocator<int> > &>,
      range_iterator_t<std::vector<int, std::allocator<int> > &>,
      range_iterator_t<std::vector<int, std::allocator<int> > &> >'))
    assert(z.end() - z.begin() == 4);
           ~~~~~~~ ^ ~~~~~~~~~
/usr/include/assert.h:89:5: note: expanded from macro 'assert'
  ((expr)                                                               \
    ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.1/../../../../include/c++/5.4.1/bits/stl_bvector.h:208:3: note:
      candidate function not viable: no known conversion from 'sentinel' (aka
      'zip_with_sentinel<agency::experimental::detail::forward_as_tuple_functor,
      range_sentinel_t<std::vector<int, std::allocator<int> > &>,
      range_sentinel_t<std::vector<int, std::allocator<int> > &>,
      range_sentinel_t<std::vector<int, std::allocator<int> > &> >') to 'const
      std::_Bit_iterator_base' for 1st argument
  operator-(const _Bit_iterator_base& __x, const _Bit_iterator_base& __y)
  ^
../../../agency/experimental/ranges/zip_with.hpp:204:17: note: candidate template
      ignored: couldn't infer template argument 'OtherIterator'
    friend auto operator-(const zip_with_iterator& lhs, const zip_with_iterator<O...
                ^
../../../agency/experimental/ranges/zip_with.hpp:332:17: note: candidate template
      ignored: couldn't infer template argument 'OtherIterator'
    friend auto operator-(const zip_with_sentinel& lhs, const zip_with_iterator<O...
                ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.1/../../../../include/c++/5.4.1/bits/stl_iterator.h:333:5: note:
      candidate template ignored: could not match 'reverse_iterator' against
      'zip_with_sentinel'
    operator-(const reverse_iterator<_Iterator>& __x,
    ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.1/../../../../include/c++/5.4.1/bits/stl_iterator.h:387:5: note:
      candidate template ignored: could not match 'reverse_iterator' against
      'zip_with_sentinel'
    operator-(const reverse_iterator<_IteratorL>& __x,
    ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.1/../../../../include/c++/5.4.1/bits/stl_iterator.h:918:5: note:
      candidate template ignored: could not match '__normal_iterator' against
      'zip_with_sentinel'
    operator-(const __normal_iterator<_IteratorL, _Container>& __lhs,
    ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.1/../../../../include/c++/5.4.1/bits/stl_iterator.h:930:5: note:
      candidate template ignored: could not match '__normal_iterator' against
      'zip_with_sentinel'
    operator-(const __normal_iterator<_Iterator, _Container>& __lhs,
    ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.1/../../../../include/c++/5.4.1/bits/stl_iterator.h:1145:5: note:
      candidate template ignored: could not match 'move_iterator' against
      'zip_with_sentinel'
    operator-(const move_iterator<_IteratorL>& __x,
    ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.1/../../../../include/c++/5.4.1/bits/stl_iterator.h:1152:5: note:
      candidate template ignored: could not match 'move_iterator' against
      'zip_with_sentinel'
    operator-(const move_iterator<_Iterator>& __x,
    ^
zip.cpp:90:25: error: invalid operands to binary expression ('sentinel' (aka
      'zip_with_sentinel<agency::experimental::detail::forward_as_tuple_functor,
      range_sentinel_t<std::vector<int, std::allocator<int> > &>,
      range_sentinel_t<std::vector<int, std::allocator<int> > &>,
      range_sentinel_t<std::vector<int, std::allocator<int> > &> >') and 'iterator' (aka
      'zip_with_iterator<agency::experimental::detail::forward_as_tuple_functor,
      range_iterator_t<std::vector<int, std::allocator<int> > &>,
      range_iterator_t<std::vector<int, std::allocator<int> > &>,
      range_iterator_t<std::vector<int, std::allocator<int> > &> >'))
    auto size = z.end() - z.begin();
                ~~~~~~~ ^ ~~~~~~~~~
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.1/../../../../include/c++/5.4.1/bits/stl_bvector.h:208:3: note:
      candidate function not viable: no known conversion from 'sentinel' (aka
      'zip_with_sentinel<agency::experimental::detail::forward_as_tuple_functor,
      range_sentinel_t<std::vector<int, std::allocator<int> > &>,
      range_sentinel_t<std::vector<int, std::allocator<int> > &>,
      range_sentinel_t<std::vector<int, std::allocator<int> > &> >') to 'const
      std::_Bit_iterator_base' for 1st argument
  operator-(const _Bit_iterator_base& __x, const _Bit_iterator_base& __y)
  ^
../../../agency/experimental/ranges/zip_with.hpp:204:17: note: candidate template
      ignored: couldn't infer template argument 'OtherIterator'
    friend auto operator-(const zip_with_iterator& lhs, const zip_with_iterator<O...
                ^
../../../agency/experimental/ranges/zip_with.hpp:332:17: note: candidate template
      ignored: couldn't infer template argument 'OtherIterator'
    friend auto operator-(const zip_with_sentinel& lhs, const zip_with_iterator<O...
                ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.1/../../../../include/c++/5.4.1/bits/stl_iterator.h:333:5: note:
      candidate template ignored: could not match 'reverse_iterator' against
      'zip_with_sentinel'
    operator-(const reverse_iterator<_Iterator>& __x,
    ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.1/../../../../include/c++/5.4.1/bits/stl_iterator.h:387:5: note:
      candidate template ignored: could not match 'reverse_iterator' against
      'zip_with_sentinel'
    operator-(const reverse_iterator<_IteratorL>& __x,
    ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.1/../../../../include/c++/5.4.1/bits/stl_iterator.h:918:5: note:
      candidate template ignored: could not match '__normal_iterator' against
      'zip_with_sentinel'
    operator-(const __normal_iterator<_IteratorL, _Container>& __lhs,
    ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.1/../../../../include/c++/5.4.1/bits/stl_iterator.h:930:5: note:
      candidate template ignored: could not match '__normal_iterator' against
      'zip_with_sentinel'
    operator-(const __normal_iterator<_Iterator, _Container>& __lhs,
    ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.1/../../../../include/c++/5.4.1/bits/stl_iterator.h:1145:5: note:
      candidate template ignored: could not match 'move_iterator' against
      'zip_with_sentinel'
    operator-(const move_iterator<_IteratorL>& __x,
    ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.1/../../../../include/c++/5.4.1/bits/stl_iterator.h:1152:5: note:
      candidate template ignored: could not match 'move_iterator' against
      'zip_with_sentinel'
    operator-(const move_iterator<_Iterator>& __x,
    ^
In file included from zip.cpp:6:
In file included from ../../../agency/experimental/ranges/zip.hpp:5:
../../../agency/experimental/ranges/zip_with.hpp:406:20: error: invalid operands to
      binary expression ('sentinel' (aka
      'zip_with_sentinel<agency::experimental::detail::forward_as_tuple_functor,
      range_sentinel_t<std::vector<int, std::allocator<int> > &>,
      range_sentinel_t<std::vector<int, std::allocator<int> > &> >') and 'iterator' (aka
      'zip_with_iterator<agency::experimental::detail::forward_as_tuple_functor,
      range_iterator_t<std::vector<int, std::allocator<int> > &>,
      range_iterator_t<std::vector<int, std::allocator<int> > &> >'))
      return end() - begin();
             ~~~~~ ^ ~~~~~~~
zip.cpp:165:27: note: in instantiation of member function
      'agency::experimental::zip_with_view<agency::experimental::detail::forward_as_tuple_functor,
      std::vector<int, std::allocator<int> > &, std::vector<int, std::allocator<int> >
      &>::size' requested here
    for(int i = 0; i < z1.size(); ++i)
                          ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.1/../../../../include/c++/5.4.1/bits/stl_bvector.h:208:3: note:
      candidate function not viable: no known conversion from 'sentinel' (aka
      'zip_with_sentinel<agency::experimental::detail::forward_as_tuple_functor,
      range_sentinel_t<std::vector<int, std::allocator<int> > &>,
      range_sentinel_t<std::vector<int, std::allocator<int> > &> >') to 'const
      std::_Bit_iterator_base' for 1st argument
  operator-(const _Bit_iterator_base& __x, const _Bit_iterator_base& __y)
  ^
../../../agency/experimental/ranges/zip_with.hpp:204:17: note: candidate template
      ignored: couldn't infer template argument 'OtherIterator'
    friend auto operator-(const zip_with_iterator& lhs, const zip_with_iterator<O...
                ^
../../../agency/experimental/ranges/zip_with.hpp:332:17: note: candidate template
      ignored: couldn't infer template argument 'OtherIterator'
    friend auto operator-(const zip_with_sentinel& lhs, const zip_with_iterator<O...
                ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.1/../../../../include/c++/5.4.1/bits/stl_iterator.h:333:5: note:
      candidate template ignored: could not match 'reverse_iterator' against
      'zip_with_sentinel'
    operator-(const reverse_iterator<_Iterator>& __x,
    ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.1/../../../../include/c++/5.4.1/bits/stl_iterator.h:387:5: note:
      candidate template ignored: could not match 'reverse_iterator' against
      'zip_with_sentinel'
    operator-(const reverse_iterator<_IteratorL>& __x,
    ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.1/../../../../include/c++/5.4.1/bits/stl_iterator.h:918:5: note:
      candidate template ignored: could not match '__normal_iterator' against
      'zip_with_sentinel'
    operator-(const __normal_iterator<_IteratorL, _Container>& __lhs,
    ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.1/../../../../include/c++/5.4.1/bits/stl_iterator.h:930:5: note:
      candidate template ignored: could not match '__normal_iterator' against
      'zip_with_sentinel'
    operator-(const __normal_iterator<_Iterator, _Container>& __lhs,
    ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.1/../../../../include/c++/5.4.1/bits/stl_iterator.h:1145:5: note:
      candidate template ignored: could not match 'move_iterator' against
      'zip_with_sentinel'
    operator-(const move_iterator<_IteratorL>& __x,
    ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.1/../../../../include/c++/5.4.1/bits/stl_iterator.h:1152:5: note:
      candidate template ignored: could not match 'move_iterator' against
      'zip_with_sentinel'
    operator-(const move_iterator<_Iterator>& __x,
    ^
In file included from zip.cpp:6:
In file included from ../../../agency/experimental/ranges/zip.hpp:5:
../../../agency/experimental/ranges/zip_with.hpp:406:20: error: invalid operands to
      binary expression ('sentinel' (aka
      'zip_with_sentinel<agency::experimental::detail::forward_as_tuple_functor,
      range_sentinel_t<std::vector<int, std::allocator<int> > &>,
      range_sentinel_t<std::vector<int, std::allocator<int> > &>,
      range_sentinel_t<std::vector<int, std::allocator<int> > &> >') and 'iterator' (aka
      'zip_with_iterator<agency::experimental::detail::forward_as_tuple_functor,
      range_iterator_t<std::vector<int, std::allocator<int> > &>,
      range_iterator_t<std::vector<int, std::allocator<int> > &>,
      range_iterator_t<std::vector<int, std::allocator<int> > &> >'))
      return end() - begin();
             ~~~~~ ^ ~~~~~~~
zip.cpp:183:14: note: in instantiation of member function
      'agency::experimental::zip_with_view<agency::experimental::detail::forward_as_tuple_functor,
      std::vector<int, std::allocator<int> > &, std::vector<int, std::allocator<int> > &,
      std::vector<int, std::allocator<int> > &>::size' requested here
    assert(z.size() == 4);
             ^
/usr/include/assert.h:89:5: note: expanded from macro 'assert'
  ((expr)                                                               \
    ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.1/../../../../include/c++/5.4.1/bits/stl_bvector.h:208:3: note:
      candidate function not viable: no known conversion from 'sentinel' (aka
      'zip_with_sentinel<agency::experimental::detail::forward_as_tuple_functor,
      range_sentinel_t<std::vector<int, std::allocator<int> > &>,
      range_sentinel_t<std::vector<int, std::allocator<int> > &>,
      range_sentinel_t<std::vector<int, std::allocator<int> > &> >') to 'const
      std::_Bit_iterator_base' for 1st argument
  operator-(const _Bit_iterator_base& __x, const _Bit_iterator_base& __y)
  ^
../../../agency/experimental/ranges/zip_with.hpp:204:17: note: candidate template
      ignored: couldn't infer template argument 'OtherIterator'
    friend auto operator-(const zip_with_iterator& lhs, const zip_with_iterator<O...
                ^
../../../agency/experimental/ranges/zip_with.hpp:332:17: note: candidate template
      ignored: couldn't infer template argument 'OtherIterator'
    friend auto operator-(const zip_with_sentinel& lhs, const zip_with_iterator<O...
                ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.1/../../../../include/c++/5.4.1/bits/stl_iterator.h:333:5: note:
      candidate template ignored: could not match 'reverse_iterator' against
      'zip_with_sentinel'
    operator-(const reverse_iterator<_Iterator>& __x,
    ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.1/../../../../include/c++/5.4.1/bits/stl_iterator.h:387:5: note:
      candidate template ignored: could not match 'reverse_iterator' against
      'zip_with_sentinel'
    operator-(const reverse_iterator<_IteratorL>& __x,
    ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.1/../../../../include/c++/5.4.1/bits/stl_iterator.h:918:5: note:
      candidate template ignored: could not match '__normal_iterator' against
      'zip_with_sentinel'
    operator-(const __normal_iterator<_IteratorL, _Container>& __lhs,
    ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.1/../../../../include/c++/5.4.1/bits/stl_iterator.h:930:5: note:
      candidate template ignored: could not match '__normal_iterator' against
      'zip_with_sentinel'
    operator-(const __normal_iterator<_Iterator, _Container>& __lhs,
    ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.1/../../../../include/c++/5.4.1/bits/stl_iterator.h:1145:5: note:
      candidate template ignored: could not match 'move_iterator' against
      'zip_with_sentinel'
    operator-(const move_iterator<_IteratorL>& __x,
    ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.1/../../../../include/c++/5.4.1/bits/stl_iterator.h:1152:5: note:
      candidate template ignored: could not match 'move_iterator' against
      'zip_with_sentinel'
    operator-(const move_iterator<_Iterator>& __x,
    ^
8 errors generated.
```

**Comment by jaredhoberock** [comment]
Thanks, that helps. My guess is that your `clang-3.5` (`3.5.0-4ubuntu2~trusty2`) has a bug that was fixed in my newer `clang-3.5` (`3.5.2-3ubuntu1`). There must be something about `zip_with_iterator`'s relational operators that `clang-3.5.0` doesn't like.

Fortunately, little of Agency's functionality depends on that iterator. If you're not able to update your version of clang, you can build subdirectories underneath `/testing` piecemeal. For example, you can build and run the tests underneath `/testing/bulk_invoke` with this command:

    $ scons -j8 bulk_invoke/run_tests

Finally, it's also possible to build with `g++` instead of `clang`:

    $ scons -j16 arch=sm_35 CXX=g++

**Comment by ctcyang** [comment]
Thanks for your help. I installed clang 3.5.2, which fixes the above error. This issue can be closed.
